ui/components/app/modals/transaction-already-confirmed/transaction-already-confirmed.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useContext } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getBlockExplorerLink } from '@metamask/etherscan-link';
+import { type TransactionMeta } from '@metamask/transaction-controller';
+import { type NetworkClientConfiguration } from '@metamask/network-controller';
+import { getRpcPrefsForCurrentProvider } from '../../../../selectors';
+import { useModalProps } from '../../../../hooks/useModalProps';
+
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalBody,
+  Text,
+  ModalFooter,
+} from '../../../component-library';
+import {
+  AlignItems,
+  FlexDirection,
+} from '../../../../helpers/constants/design-system';
+import { I18nContext } from '../../../../contexts/i18n';
+import { MetaMaskReduxState } from '../../../../store/store';
+
+export default function TransactionAlreadyConfirmed() {
+  const {
+    hideModal,
+    props: { originalTransactionId },
+  } = useModalProps();
+  const t = useContext(I18nContext);
+  const dispatch = useDispatch();
+
+  const { transaction, rpcPrefs } = useSelector((state: MetaMaskReduxState) => {
+    const tx: TransactionMeta = state.metamask.transactions.find(
```

Comment:
Should we use a selector like `getTransaction` to avoid the re-rendering and avoid the inline state searching?

Reply:
Done in https://github.com/MetaMask/metamask-extension/pull/22943/commits/a9a67f32546a33a7d1dd9bd25221f6cc110610e6